wandb/sdk/data_types/base_types/media.py

```diff
@@ -156,6 +157,7 @@ def to_json(self, run: Union["LocalRun", "LocalArtifact"]) -> dict:
         """
         # NOTE: uses of Audio in this class are a temporary hack -- when Ref support moves up
         # into Media itself we should get rid of them
+        from wandb import Image
```

Comment:
Notice the comment above. While it is true that this PR does address the specific ticket requirements, I am wondering if you considered how we could get away from these specific  hacks? Basically, how could we generically enable all Media classes to support refs? Maybe blindly allowing all media classes to support it is too risky? If so, could we create a mixin class that subclassers can inherit from? This would allow us to minimize any reference-specific code in the media types. Another thing to consider: what if we had a general `Path` class that supported multiple different formats and could be used in these cases to avoid any path-specific logic inside of media classes?

Again, i know this is additional scope, so not blocking, but it would be good to consider if there are general ways to both: a) improve the quality of this part of the code base; and b) cover more features in a future proof way. Consider that the next ask is to solve for Video refs - it would be great if that was a 1 liner, or even automatic!

Reply:
yeah definitely agree with this, I will create another ticket to refactor the code. I know there are more talks of supporting references like this for other media